Fix a few menu and footer issues

tzonghao opened the pull request:
- ~~Update jQuery to the latest~~
- Escape key to collapse menus
- Fix menu z-index so it's not hidden behind footer
- ~~Update example file to include footer~~ (Header script and example updated)
- ~~Push footer to the bottom, inline styles as example~~ (Add example to README.md)
- ~~Toggle menu when it loses focus~~

tzonghao committed:
Update jQuery to latest version, load from relative paths

- jquery 1.9 -> 3.2
- index.html loads main css and js files from relative paths, instead
  from Pages urls. This also enables local developments.

tzonghao committed:
Stick footer to the bottom

tzonghao committed:
Fixed: expanded menu hidden behind footer

tzonghao committed:
Fix menu expand/collapse toggle

- "esc" key collapse expanded menu and sub menus
- remove old/unused callback functions
- fix typo that prevents removing 'is-open' when toggled by others

tzonghao committed:
Update example file

tzonghao committed:
Add blur callback to toggle menu

tzonghao committed:
Remove blur callback

tzonghao committed:
Add "mouseleave" callback to close menu

tzonghao committed:
Remove menu slide down when toggle regains focus

tzonghao committed:
Toggle to close menu when mouse leaves menu

jonraedeke reviewed (commented):
@mdsteveb What do you think about making the css and js non-dynamic? I think might cause stale assets to get out of sync with the dynamic markup.

Also, any changes to the index.html head should also be made to the nist_footer.html and nist_header_bar.html files as well.

jonraedeke commented on index.html:
```diff
@@ -1,10 +1,25 @@
   <link rel="stylesheet" href="css/nist-combined.css">
+
+  <style>
+    /* push footer to the bottom */
```
I think I'd rather not have inline css here since it complicates this simple template. I haven't tested flexbox on IE11 extensively, but this is a nice trick. What about including snippets in the README.md file for various tips and tricks?

tzonghao replied:
I agree that it complicates the simple template. How about we put the snippet in a paragraph in the main content (this file) so it's more visible to users, instead of in the README only visible on GitHub? ~~Let me know either way, then I'll make the change.~~ I moved the example to README for now.

jonraedeke commented on js/nist-header-footer.js:
```diff
@@ -78,6 +78,8 @@ $(document).ready(function(){
         $('.nist-header__nav-inner').slideDown(300);
       });
 
+      $('.nist-header__nav-inner .nist-menu').mouseleave(toggleMenu);
```
This causes a weird behavior when a menu item is expanded. When you hover past the open item, the entire menu collapses.

tzonghao replied:
I'll remove this.

jonraedeke reviewed (changes requested):
@tzonghao Thanks for you work on this, especially on the js! Just a few changes noted in the code. Hopefully I comment in the right spot.

1. I don't think we want js and css non-dynamic for now since we don't really have releases.
2. I prefer the inline CSS to be a snippet in the README.md. Or maybe there is a better way.
3. When a menu item is expanded, hovering past it closes the entire menu

tzonghao committed:
Remove mouse leave callback

tzonghao committed:
Move flexbox example to README

tzonghao committed:
Restore and update CSS/JS and examples

- Restore original full URLs instead of relative URLs
- Get nist-header.js up to date

tzonghao commented:
@jonraedeke I've pushed more changes based on your feedback. Thanks.

tzonghao commented:
@jonraedeke Just want to see if there's anything else you'd want to change. I have another PR I'd like to submit, potentially.
